examples/pypi_wheel_provider.py

```diff
@@ -123,19 +123,20 @@ def get_base_requirement(self, candidate):
     def get_preference(self, resolution, candidates, information):
         return len(candidates)
 
-    def find_matches(self, requirement):
+    def iter_matches(self, requirements):
         candidates = []
-        name = requirement.name
-        extras = requirement.extras
+        name = requirements[0].name
+        extras = requirements[0].extras
 
         # Need to pass the extras to the search, so they
         # are added to the candidate at creation - we
         # treat candidates as immutable once created.
         for c in get_project_from_pypi(name, extras):
             version = c.version
-            if version in requirement.specifier:
+            if all(version in r.specifier for r in requirements):
                 candidates.append(c)
-        return candidates
+        for c in reversed(candidates):
```

Comment:
Maybe do the `reversed()` on the call to `get_project_from_pypi`? That call itself depends on the order of elements in the index, so maybe `sorted(get_project_from_pypi(name, extras), key=operator.attrgetter('version'), reversed=True)` would be better still. Then we can just yield c in the loop, rather than building a list of candidates.

Reply:
Going with the `sorted()` approach.